pandas/types/cast.py

```diff
@@ -527,8 +527,9 @@ def _astype_nansafe(arr, dtype, copy=True):
     elif (np.issubdtype(arr.dtype, np.floating) and
           np.issubdtype(dtype, np.integer)):
 
-        if np.isnan(arr).any():
-            raise ValueError('Cannot convert NA to integer')
+        if not np.isfinite(arr).all():
+            raise ValueError('Cannot convert non-finite values to integer')
```

Comment:
Yeah, no problem. I debated adding it, but went for succinctness. I agree it is informative, adding it now.